fe/src/pages/campaigns/[id].vue

```diff
@@ -40,20 +47,81 @@ import { storeToRefs } from "pinia";
 import { toast } from "vue3-toastify";
 import { ethers } from "ethers";
 import CampaignProps from "~/types/CampaignProps";
-import CampaignSample from "~/mocks/campaign-sample.json";
 import { useWalletStore } from "~/store/wallet";
 const route = useRoute();
-const { truncate } = useUtils();
+const { truncate, getDaysLeft } = useUtils();
 
 const { $getSmartContract: getSmartContract } = useNuxtApp();
 
-const campaignValueSample = ref<CampaignProps>(CampaignSample);
-
 const isLoading = ref<boolean>(false);
-
+const canEdit = ref<boolean>(false);
 const useWallet = useWalletStore();
-const { isConnected } = storeToRefs(useWallet);
+const { isConnected, address } = storeToRefs(useWallet);
 const { id } = route.params;
+const campaign = ref<CampaignProps>({
+  campaignId: 0,
+  title: "",
+  creator: {
+    address: "",
+    fullName: "",
+    imgSrc: "",
+  },
+  imgSrc: "",
+  story: "",
+  daysLeft: 0,
+  totalDonation: 0,
+  totalBackers: 0,
+  donations: [],
+});
+
+const getCampaign = async (id: number): Promise<void> => {
+  isLoading.value = true;
+  const smartContract = await getSmartContract();
+  if (smartContract !== null) {
+    smartContract
+      .getCampaign(id)
+      .then((result) => {
```

Comment:
seems there is a lot of logics, im thinking to use try catch will be cleaner. 

const getCampaign = (data) => {
 return  {
          campaignId: Number(data[0][0]),
          title: data[0][3],
          creator: {
            address: data[0][1],
            fullName: data[0][2],
            imgSrc:
              "https://img.freepik.com/free-psd/3d-illustration-person_23-2149436182.jpg",
          },
          imgSrc:
            "https://images.unsplash.com/photo-1529390079861-591de354faf5",
          story: data[0][4],
          daysLeft: getDaysLeft(data[0][7]),
          totalDonation: Number(ethers.formatEther(data[0][6])),
          totalBackers: Number(data[0][8]),
          donations: [],
        };
}

try {
  const result = smartContract.getCampaign(id)
  const campaign = getCampaign(result)
} catch {

}

Reply:
See new commit: [Restructure code](https://github.com/framgia/sph-solidity/pull/36/commits/a633718a2bb99fd10727f521e9307acba65b8bf7)